tests/integration/_validator/testcases/test_validator.go

```diff
@@ -666,3 +732,12 @@ func (t *ValidatorTest) updateTestData(tokens map[string]string) error {
 func (t *ValidatorTest) filePath(file string) string {
 	return fmt.Sprintf("%s/%s/%s", file_utils.ValidatorTestCasesPath(), "data", file)
 }
+
+func (t *ValidatorTest) overrideMaasClient() {
+	clouds.GetMaasClient = func(maasURL, maasToken string) (*maasclient.Client, error) {
```

Comment:
You should preserve the original value of `clouds.GetMaasClient` by pushing it into the test context up front, then reset it during `TearDown`. Take a look at palette-cli's vSphere tests to see an example of what I'm describing.

What you have is fine, but it isn't idempotent. Any subsequent test suite would end up using the overridden function for `clouds.GetMaasClient`, which is not ideal.